src/commands/serve.rs

```diff
@@ -450,7 +450,17 @@ async fn handle_request(
     match receiver.await {
         Ok(Ok(resp)) => Ok(resp),
         Ok(Err(e)) => Err(e.into()),
-        Err(_) => bail!("guest never invoked `response-outparam::set` method"),
+        Err(_) => {
+            // If the receiver has an error (`RecvError`), it will not describe
+            // the real failure in any meaningful way. Instead we retrieve and
+            // propagate the error from inside the task which should more
+            // clearly tell the user what went wrong.
+            let e = match task.await {
```

Comment:
Do we need to bound the amount of time spent here, or are we guaranteed that the result should be available immediately because the `receiver.await` failed?

Reply:
As I looked at it, I concluded that yes, if the receiver failed it is because the task failed. But maybe I missed something or perhaps future changes will invalidate that. But in that case, `wasmtime serve` will be dumping errors to stderr anyways so it should be apparent to users that something is wrong apart from the (hypothetical) additional slowdown. And they'll probably care more about whatever caused the task to fail.

Reply:
Ok, added a comment about this assumption: f09760e